core/extrinsics_submission_service/json_transport.hpp

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_EXTRINSICS_SUBMISSION_SERVICE_JSON_TRANSPORT_HPP
+#define KAGOME_CORE_EXTRINSICS_SUBMISSION_SERVICE_JSON_TRANSPORT_HPP
+
+#include <boost/signals2/signal.hpp>
+#include <outcome/outcome.hpp>
+
+namespace kagome::service {
+  class JsonTransport {
+   public:
+    using OnData = boost::signals2::signal<void(const std::string &)>;
+    using OnResponse = boost::signals2::slot<void(const std::string &)>;
+
+    virtual ~JsonTransport() = default;
+
+    explicit inline JsonTransport(
+        std::function<void(const std::string &)> response_callback)
```

Comment:
define using on `void(const std::string&)` and use it for signal, slot and here

Reply:
done